pg2kinesis/stream.py

```diff
@@ -33,8 +34,22 @@ def put_message(self, fmt_msg):
         agg_record = None
 
         if fmt_msg:
-            agg_record = self._record_agg.add_user_record(fmt_msg.change.xid, fmt_msg.fmt_msg)
+            if fmt_msg.is_bulk:
+                for change in fmt_msg.change.changes:
+                    agg_record = self._record_agg.add_user_record(fmt_msg.change.xid, fmt_msg.fmt_msg + json.dumps({
+                        'xid': fmt_msg.change.xid,
+                        'change': change,
+                    }))
+                    agg_record = self._check_full_agg_record(agg_record)
```

Comment:
This line which appears in all 3 branches can be moved before the return, and removed from the if all together.

Reply:
One branch needs to check if the agg is full as it loops over changes so that won't quite work. The more I look at this though I wonder if it would be cleaner to move the loop outside of here altogether and do this work up in Consume.